### DISCUSSION TITLE
Diagram generation

### DISCUSSION THREAD

**Discussion by Serhioromano** [eclipse-langium/langium#824] [Ideas]
Chevrotain has a great built in diagram generator. All you need is to serialize parser.

```js
const serializedGrammar = parser.getSerializedGastProductions();
const htmlText = chevrotain.createSyntaxDiagramsCode(serializedGrammar);
const outPath = path. Resolve(__dirname, "./")
fs.writeFileSync(outPath + "/generated_diagrams.html", htmlText);
```

Can we add options `-d` parameter for `langium generate` that could also generate that diagram to see how it actually built?

**Comment by spoenemann** [comment]
👍 I'd make this a separate command like `langium syntax-diagram`

**Reply by msujew** [reply]
Why not have it as a config parameter like for syntax highlighting?

**Reply by Serhioromano** [reply]
I like the direction of this conversation. For me it is not a critical how. Separate or parameter the fact that it is there will be much appreciated.

**Reply by spoenemann** [reply]
> Why not have it as a config parameter like for syntax highlighting?

I thought you typically don't need that to be always up-to-date, but only when you want to debug the grammar. On the other hand, it could be useful to always generate it to include it in documentation. In that case adding this to the config makes sense.
